lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -208,15 +208,27 @@ combineDbAndLsqData
     -> Map PoolId PoolDbData
     -> Map PoolId Api.ApiStakePool
 combineDbAndLsqData sp =
-    Map.merge lsqButNoChain chainButNoLsq bothPresent
+    Map.merge lsqButNoDb dbButNoLsq bothPresent
   where
-    lsqButNoChain = traverseMissing $ \k lsq -> pure $ mkApiPool k lsq Nothing
-
-    -- In case our chain following has missed a retirement certificate, we
-    -- treat the lsq data as the source of truth, and dropMissing here.
-    chainButNoLsq = dropMissing
+    lsqButNoDb = traverseMissing $ \k lsq ->
```

Comment:
Should we not rely 100% on the DB now, and `dropMissing` on `lsqButNoDb`?

Reply:
> Should we not rely 100% on the DB now, and `dropMissing` on `lsqButNoDb`?

We could indeed do that. If we did, we'd be able to guarantee the presence of the following fields for each `ApiStakePool` object we produce:

- `cost`
- `margin`
- `pledge`

(See: https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/src/Cardano/Wallet/Api/Types.hs#L406-L408)

We'd then be able to make the following simplification:
```patch
@@ -403,9 +403,9 @@ data ApiStakePool = ApiStakePool
     { id :: !(ApiT PoolId)
     , metrics :: !ApiStakePoolMetrics
     , metadata :: !(Maybe (ApiT StakePoolMetadata))
-    , cost :: !(Maybe (Quantity "lovelace" Natural))
-    , margin :: !(Maybe (Quantity "percent" Percentage))
-    , pledge :: !(Maybe (Quantity "lovelace" Natural))
+    , cost :: !(Quantity "lovelace" Natural)
+    , margin :: !(Quantity "percent" Percentage)
+    , pledge :: !(Quantity "lovelace" Natural)
     , retirement :: !(Maybe ApiEpochInfo)
     } deriving (Eq, Generic, Show)
```

Reply:
> Should we not rely 100% on the DB now, and dropMissing on lsqButNoDb?

Fixed in 457ad036c1e0f7f93e637c8453e6ec5f49c9153c.

Reply:
> We'd then be able to make the following simplification:
> ```patch
> @@ -403,9 +403,9 @@ data ApiStakePool = ApiStakePool
>      { id :: !(ApiT PoolId)
>      , metrics :: !ApiStakePoolMetrics
>      , metadata :: !(Maybe (ApiT StakePoolMetadata))
> -    , cost :: !(Maybe (Quantity "lovelace" Natural))
> -    , margin :: !(Maybe (Quantity "percent" Percentage))
> -    , pledge :: !(Maybe (Quantity "lovelace" Natural))
> +    , cost :: !(Quantity "lovelace" Natural)
> +    , margin :: !(Quantity "percent" Percentage)
> +    , pledge :: !(Quantity "lovelace" Natural)
>      , retirement :: !(Maybe ApiEpochInfo)
>     } deriving (Eq, Generic, Show)
> ```

Fixed in:
- efbfd59beee7bc3feac20c0d3d365c073a5c45b4 (Simplify field types for `ApiStakePool`)
- 4a0274329190c6c8fba9a64a2d063028c3782c14 (Mark fields as required in specification for `ApiStakePool`)
- bec56c630fa48a0614b39f8ce2b8b56ed146eb46 (Regenerate golden test data for `ApiStakePool`)